### PR TITLE
2022-11 CWG Motion 9

### LINKED ISSUE #5964: [2022-11 CWG Motion 9] P2564R3 consteval needs to propagate up

### PR DESCRIPTION
**Pull request by tkoeppe**
P2564R3 consteval needs to propagate up

Fixes #5964.
Fixes cplusplus/nbballot#467
Fixes cplusplus/papers#1223

### CODE REVIEW

File: source/expressions.tex [hunk 1]
```diff
@@ -7839,7 +7833,112 @@
...
-An immediate invocation shall be a constant expression.
+An aggregate initialization is an immediate invocation
+if it evaluates a default member initializer
+that has a subexpression that is an immediate-escalating expression.
+
+\pnum
+\indexdefn{expression!immediate-escalating}%
+\indexdefn{conversion!immediate-escalating}%
+\indexdefn{immediate-escalating!expression|see{expression, immediate-escalating}}%
+\indexdefn{immediate-escalating!conversion|see{conversion, immediate-escalating}}%
```
[the remark this answers is not in the document]
Comment (tkoeppe) [reply source/expressions.tex:7844]: Note also the index entry for "immediately-escalating function" below.

[the remark this answers is not in the document]
Comment (JohelEGP) [reply source/expressions.tex:7844]: Nothing from the top of my mind.

I usually look at `macros.tex` to see if anything does a better job. Another method I use is looking for existing wording to see how it's done in hopes of finding a better formulation.

[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/expressions.tex:7844]: @tkoeppe 

\defnadj{immediate-escalating}{expression}

should be your one-stop shop here.

[the remark this answers is not in the document]
Comment (tkoeppe) [reply source/expressions.tex:7844]: @jensmaurer Not quite, because that also prints output, right?

[the remark this answers is not in the document]
Comment (JohelEGP) [reply source/expressions.tex:7844]: Using `\defnadj` requires tweaking the wording. It also undoes the nesting of the adjective in the index:

One option I see is defining `\indexadj`, which does the indexing part of `\defnadj`.

**Review by burblebee — APPROVED** [review]

**Pull request merged**